EVE9: 5 Most Recent Pull Requests

haydenudelson opened the pull request:
#### What does this PR do?
This PR shows the 5 most recent Pull Request's for the DDF repo.  It shows the title, number, author, and date created of each PR.  From here, it would be easy to switch which repo to display based on team-specific needs.

#### Who is reviewing it?
@vinamartin @Kjames5269 @bankent1 @alchucam @mcalcote

#### How should this be tested?
To test this, you'll need the GitHub API token, which I can provide in the Slack Channel.  Then, simply "yarn build" and "yarn start".

#### Screenshots
<img width="577" alt="githubAPI" src="https://user-images.githubusercontent.com/20527865/60736150-d3001c80-9f0a-11e9-9449-078f47aaf052.PNG">

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist.
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

vinamartin commented:
Add something to the README similar to the setting up slack section: https://github.com/connexta/eve#setting-up-slack

alchucam commented:
![image](https://user-images.githubusercontent.com/40583917/60738421-22e2e180-9f13-11e9-810e-3d09325d19f6.png)
Getting this error during testing in google chrome environment!

vinamartin commented:
Viewing at 100% on my tiny monitor
<img width="183" alt="Screen Shot 2019-07-05 at 12 05 52" src="https://user-images.githubusercontent.com/9013407/60741163-4448cb00-9f1d-11e9-9994-9117d01f0d24.png">

Viewing at 90% on my tiny monitor
<img width="224" alt="Screen Shot 2019-07-05 at 12 06 27" src="https://user-images.githubusercontent.com/9013407/60741207-63dff380-9f1d-11e9-9664-506e3e8a257a.png">

This might just be my display, but the spacing seems off here. It gets worse the more I zoom out too.
<img width="1028" alt="Screen Shot 2019-07-05 at 12 07 32" src="https://user-images.githubusercontent.com/9013407/60741238-7d813b00-9f1d-11e9-8b03-b4e310c01e36.png">

How do you have the spacing now? Is it possible to make it scale better?

haydenudelson commented:
Thanks for the feedback- I was able to address the CORS issue but had to change how we authenticated when calling the API.  The Spacing issue was because the widgets were set to maximize space in-between, so I changed it to keep items stored near the top.  This might be a design decision to discuss sometime this sprint as we work on refining the UI. Updated the README too!

Kjames5269 commented on src/githubCaller.js:
```diff
@@ -0,0 +1,131 @@
+
+  componentDidMount() {
+    this.callGithub();
+    setInterval(() => this.callGithub(), CALL_FREQ);
```
❗️ This interval should be cleared within a `componentWillUnmount` function

haydenudelson commented on src/githubCaller.js:
```diff
@@ -0,0 +1,131 @@
+const CLIENT_ID = process.env.GITHUB_CLIENT_ID;
+const CLIENT_SECRET = process.env.GITHUB_CLIENT_SECRET;
+
+const Box = styled.div`
```
I could make an index.css file that holds the styling for all our components so we have one place to draw from, but I'm not sure if that's in the scope of this PR.  What do you think?

Kjames5269 replied:
I also think it's scope creep 🙂 but it might be something to start thinking about

Kjames5269 commented on src/githubCaller.js:
```diff
@@ -115,7 +118,13 @@ export default class Github extends React.Component {
   componentDidMount() {
     this.callGithub();
-    setInterval(() => this.callGithub(), CALL_FREQ);
+    this.setState({
```
✏️ I don't think this needs to be in state. I believe you should just be able to do `this.interval = setInterval()`

Kjames5269 commented on src/githubCaller.js:
```diff
@@ -67,24 +76,21 @@ export default class Github extends React.Component {
+      <div key={i}>
+        <Octicon icon={GitPullRequest} size="medium" />
+        <PRMainLine>
+          <PRTitle>{this.state.data[i].title}</PRTitle>
```
✏️ `<PRTitle>{pr.title}</PRTitle>`
